src/strategies/libs/StrategyLib.sol

```diff
@@ -55,14 +55,24 @@ library StrategyLib {
         address receiver
     ) external returns (uint[] memory amountsOut) {
         uint len = assets.length;
-        amountsOut = new uint[](len);
+        amountsOut = previewWithdraw(assets, amount, total_);
         for (uint i; i < len; ++i) {
-            amountsOut[i] = IERC20(assets[i]).balanceOf(address(this)) * amount / total_;
-            amountsOut[i] = IERC20(assets[i]).balanceOf(address(this)) * amount / total_;
             IERC20(assets[i]).transfer(receiver, amountsOut[i]);
         }
     }
 
+    function previewWithdraw(
+        address[] memory assets,
+        uint amount,
+        uint total_
+    ) public view returns (uint[] memory amountsOut) {
+        uint len = assets.length;
+        amountsOut = new uint[](len);
+        for (uint i; i < len; ++i) {
+            amountsOut[i] = balance(assets[i]) * amount / total_;
```

Comment:
mistake
assets can be invested,, not at strategy contract balance
use assetsAmounts() instead balance